src/nvim/fileio.c

```diff
@@ -4926,8 +4956,14 @@ int buf_check_timestamp(buf_T *buf)
         xstrlcat(tbuf, "\n", tbuf_len - 1);
         xstrlcat(tbuf, mesg2, tbuf_len - 1);
       }
-      if (do_dialog(VIM_WARNING, (char_u *) _("Warning"), (char_u *) tbuf,
-                    (char_u *) _("&OK\n&Load File"), 1, NULL, true) == 2) {
+      int choice = do_dialog(VIM_WARNING, (char_u *)_("Warning"),
+                             (char_u *)tbuf,
+                             (char_u *)_("&OK\n&Show diff\n&Load File"),
+                             1, NULL, true);
+      if (choice == 2) {
+        create_buf_diff_split(buf);
+        notified_diff = true;
+      } else if (choice == 3) {
```

Comment:
I feel like show diff should be option 3 for consistency reasons.  As a user might already be used to meanings of 1/2..

Reply:
Umm, since we have to type in the first letter for choosing an option, I don't know if it is that confusing? The user could still press `o` and `l` for the old meanings. I wanted to keep the `Load File` as the last option for the prompt.